pack.go

```diff
@@ -70,14 +71,21 @@ func extractPackagedBuildpacks(ctx context.Context, pkgImageRef string, fetcher
 				},
 			}
 
-			if desc.Info == md.BuildpackInfo {
+			if desc.Info.Match(md.BuildpackInfo) { // This is the order buildpack of the package
 				mainBP = dist.BuildpackFromTarBlob(desc, b)
 			} else {
 				depBPs = append(depBPs, dist.BuildpackFromTarBlob(desc, b))
 			}
 		}
 	}
 
+	if mainBP == nil {
```

Comment:
Validate that `mainBP` is always set. Certain cases in the loop where the comparison was failing (in the first branch of https://github.com/buildpacks/pack/pull/506/files#diff-691c68b2672f8b4e9127e3935dfe84efR74-R78) caused `mainBP` to return as nil, which caused panics later down the line.